test/e2e/aws_util_test.go

```diff
@@ -0,0 +1,177 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// creates an ec2 client using aws credentials and region.
+func createEC2ServiceClient(t *testing.T, infraConfig configv1.Infrastructure, kclient client.Client) *ec2.EC2 {
+	decodedAccessKeyID, decodedSecretAccessKey, err := getIDAndKey(kclient)
+	if err != nil {
+		t.Fatalf("access key id and secret access key not create due to error: %v", err)
+	}
+	region, err := getRegion(infraConfig)
+	if err != nil {
+		t.Fatalf("region not created due to error: %v", err)
+	}
+	// Set up an AWS session
+	sess, err := createSession(region, decodedAccessKeyID, decodedSecretAccessKey)
+	if err != nil {
+		t.Fatalf("session not created due to error: %v", err)
+	}
+	// Create an EC2 service client
+	svc := ec2.New(sess)
+	return svc
+}
+
+// getClusterName fetches cluster name from the infrastructures.config.openshift.io cluster object.
+func getClusterName(infraConfig configv1.Infrastructure) (string, error) {
+	var clusterName string
+	if len(infraConfig.Status.InfrastructureName) != 0 {
+		clusterName = infraConfig.Status.InfrastructureName
+	} else {
+		return "", fmt.Errorf("cluster name not found")
+	}
+	return clusterName, nil
+}
+
+// getRegion fetches region from the infrastructures.config.openshift.io cluster object.
+func getRegion(infraConfig configv1.Infrastructure) (string, error) {
+	var region string
+	if infraConfig.Status.PlatformStatus.AWS != nil && len(infraConfig.Status.PlatformStatus.AWS.Region) != 0 {
+		region = infraConfig.Status.PlatformStatus.AWS.Region
+	} else {
+		return "", fmt.Errorf("region not found")
+	}
+	return region, nil
+}
+
+// AWSCloudCredSecretName fetches secrets for AWS from kube-system namespace.
+func AWSCloudCredSecretName() types.NamespacedName {
+	return types.NamespacedName{Namespace: "kube-system", Name: "aws-creds"}
+}
+
+// getIDAndKey fetches the aws encoded credentials and returns the decoded value.
+func getIDAndKey(kubeClient client.Client) (string, string, error) {
+	awsSecret := corev1.Secret{}
+	var decodedAccessKeyID, decodedSecretAccessKey string
+	var err error
+	if err = kubeClient.Get(context.TODO(), AWSCloudCredSecretName(), &awsSecret); err != nil {
+		return "", "", fmt.Errorf("failed to get secret: %w", err)
+	}
+	if aws_access_key_id, ok := awsSecret.Data["aws_access_key_id"]; ok || (ok && len(aws_access_key_id) != 0) {
```

Comment:
first check if the keys exists , if yes then check the length.

Reply:
I think the whole `if` doesn't make much sense if `aws_access_key_id` key doesn't exist or empty `accessKeyID` would be empty too. So I think it can be simplified to:
```go
accessKeyID = string(awsSecret.Data["aws_access_key_id"])
```